### PR TITLE
Add yaml support for custom executables and executable order control

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge adds the ability to define custom executables in the YAML workspace configuration file.

Additionally, this adds the ability to control the executables used for creating the '{command}' variable within experiments.

### CODE REVIEW

File: lib/ramble/ramble/workspace/workspace.py [hunk 1]
```diff
@@ -782,8 +785,8 @@ def all_specs(self):
         import ramble.spec
 
         specs = []
-        for app, workloads, _, _ in self.all_applications():
-            for workload, _, _, _ in self.all_workloads(workloads):
+        for app, workloads, _, _, _ in self.all_applications():
```
Comment (rfbgo) [inline lib/ramble/ramble/workspace/workspace.py:788]: Should this be `*_` too?

File: lib/ramble/ramble/workspace/workspace.py [hunk 2]
```diff
@@ -1074,7 +1094,7 @@ def concretize(self):
         mpi_dict = spack_dict[namespace.mpi_lib]
         applications_dict = spack_dict[namespace.application]
 
-        for app_name, _, _, _ in self.all_applications():
+        for app_name, _, _, _, _ in self.all_applications():
```
Comment (rfbgo) [inline lib/ramble/ramble/workspace/workspace.py:1097]: Same *_ here?

File: lib/ramble/ramble/workspace/workspace.py [hunk 3]
```diff
@@ -1688,6 +1709,14 @@ def get_workspace_env_vars(self):
             if namespace.env_var in \
             workspace_dict[namespace.ramble] else None
 
+    def get_workspace_internals(self):
```
Comment (rfbgo) [inline lib/ramble/ramble/workspace/workspace.py:1712]: these `get_workspace_xxx` blocks are very similar, do you think we should abstract the out ?

**Review by rfbgo — APPROVED** [review]
LGTM. Great to see how this was actually simple at it's core (but required adding lots of small changes)

File: lib/ramble/ramble/application.py [hunk 4]
```diff
@@ -276,9 +283,24 @@ def add_expand_vars(self, workspace):
         - command: set to the commands needed to execute the experiment
         - spack_setup: set to an empty string, so spack applications can override this
         """
+        from ramble.workspace import namespace
```
[the remark this answers is not in the document]
Comment (douglasjacobsen) [reply lib/ramble/ramble/application.py]: I hoisted it. Thanks

File: lib/ramble/ramble/workspace/workspace.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/workspace/workspace.py]: Nice catch, thanks!

File: lib/ramble/ramble/workspace/workspace.py (hunk 2 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/workspace/workspace.py]: Done

File: lib/ramble/ramble/workspace/workspace.py (hunk 3 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/workspace/workspace.py:1712]: Done!

**Review by rfbgo — APPROVED** [review]
Looks good! Thanks for the changes

**Pull request merged**